src/storage/query/QueryEdgePropsProcessor.cpp

```diff
@@ -27,16 +27,32 @@ kvstore::ResultCode QueryEdgePropsProcessor::collectEdgesProps(
         return ret;
     }
 
+    auto schema = this->schemaMan_->getEdgeSchema(spaceId_, std::abs(edgeKey.edge_type));
     // Only use the latest version.
     if (iter && iter->valid()) {
         RowWriter writer(rsWriter.schema());
         PropsCollector collector(&writer);
         auto reader = RowReader::getEdgePropReader(schemaMan_,
                                                    iter->val(),
                                                    spaceId_,
-                                                   edgeKey.edge_type);
-        this->collectProps(reader.get(), iter->key(), props, nullptr, &collector);
-        rsWriter.addRow(writer);
+                                                   std::abs(edgeKey.edge_type));
+
+        // Check if ttl data expired
+        auto retTTLOpt = getEdgeTTLInfo(edgeKey.edge_type);
```

Comment:
We don't need to do it for every EdgeKey

Reply:
Yeah.  Here is in `if`  case , which is actually only executed once.
I will do it first.